File: backend/src/main/java/com/pgvaale/backend/entity/Maid.java
package com.pgvaale.backend.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.*;

/** MAID ENTITY */
@Entity
@Table(name = "maids")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Maid extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "maid_seq")
    @SequenceGenerator(name = "maid_seq", sequenceName = "maid_sequence", initialValue = 2000, allocationSize = 1)
    private Long id;

    @NotBlank
    @Size(min = 10, max = 10)
    @Column(name = "phone_number")
    private String phoneNumber;

    @NotBlank
    @Size(min = 12, max = 12)
    @Column(name = "aadhaar")
    private String aadhaar;

    @Column(name = "services")
    private String services; // CSV or JSON ("Mobbing,Cooking")

    @Column(name = "monthly_salary")
    private double monthlySalary;

    @Column(name = "gender")
    private String gender;

    @Column(name = "timing")
    private String timing;

    @Column(name = "region")
    private String region;
    
    @Column(name = "approved")
    @Builder.Default
    private boolean approved = false; // Admin approval status
} 

File: PGVaaleDotNetBackend/Entities/Maid.cs
using System.ComponentModel.DataAnnotations;

namespace PGVaaleDotNetBackend.Entities
{
    public class Maid : BaseEntity
    {
        [Key]
        public long Id { get; set; }

        [Required]
        [StringLength(10, MinimumLength = 10)]
        public string PhoneNumber { get; set; } = string.Empty;

        [Required]
        [StringLength(12, MinimumLength = 12)]
        public string Aadhaar { get; set; } = string.Empty;

        public string Services { get; set; } = string.Empty; // CSV or JSON ("Mobbing,Cooking")

        public double MonthlySalary { get; set; }

        public string Gender { get; set; } = string.Empty;

        public string Timing { get; set; } = string.Empty;

        public string Region { get; set; } = string.Empty;
        
        public bool Approved { get; set; } = false; // Admin approval status
    }
}